### ISSUE TITLE
inline-update of autogold.Value with populated *string inside got inserts invalid Go code

### ISSUE THREAD

**Issue by fabstu** [hexops/autogold#4]
Hello! I've got a got-struct which contains string-pointers. Using inline-update, it adds the following lines to the struct:

```
MobileCountryCode: &"xxx",
MobileNetworkCode: &"xx",
```

But that is not valid golang code, so it fails to build. Can this be fixed somehow? Thank you! I'm thinking maybe add autogold.String which returns a *string of the passed-in string?

A complete repro is the following. `go test -update` inserts the line `{name: "good", want: autogold.Want("good", &A{A: &"abc", B: "def"})},`, which is no valid Go code.

```
func TestGet(t *testing.T) {
	tests := []struct {
		name string
		want autogold.Value
	}{
		{name: "good", want: autogold.Want("good", nil)},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := Get()
			tt.want.Equal(t, got)
		})
	}
}

type A struct {
	A *string
	B string
}

func Get() *A {
	str := "abc"

	return &A{
		A: &str,
		B: "def",
	}
}
```

**Comment by slimsag** [comment]
Thanks for filing! And yes, I think it should be possible to fix this - I'll take a look soon

**Comment by slimsag** [comment]
@fabstu @jan-herout I've fixed this in [Valast 1.2.0](https://github.com/hexops/valast/commit/195033de249b67b020f48c451898d779f561a6aa), with extensive handling for this including all builtin (`&"foo"`, `&23`, `&true`) and non-builtin (`&myBool(true)`) types. It should work really well now on pointers.

For example `&"foo"` will now appear as `valast.Addr("foo").(*string)` in your code.

You can get the new version in your project using:

```
go get -u github.com/hexops/autogold@v1.1.0
```

Please let me know if it works!

**Comment by fabstu** [comment]
Works well, thank you! It's sad that this is necessary, but what don't we do in Go to avoid ambiguity. 👍
